src/main/java/com/epam/task/exceptions/Purchase.java

```diff
@@ -0,0 +1,61 @@
+package com.epam.task.exceptions;
+
+import com.epam.task.exceptions.customExc.NonPositiveArgumentException;
+
+public class Purchase implements  Comparable<Purchase>{
+    protected final Product product;
+    protected final int purchasedNum;
```

Comment:
```suggestion
    private final Product product;
    private final int purchasedNum;
```
because this class is not abstract.